Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts

```diff
@@ -95,4 +158,60 @@ export class QuestionsDashboardComponent implements OnInit {
     deleteQuestionDialog() {
         this.dialog.open(DeleteQuestionDialogComponent);
     }
-}
+
+    /**
+     * Filter Questions depending on DifficultyLevel, Category and QuestionDetails. 
+     * @param question
+     * @param category
+     * @param difficultyLevel
+     * @param matchString
+     */
+    filterQuestion(question: Question[], category: Category = null, difficultyLevel: DifficultyLevel = null, matchString: string = null): Question[] {
+        let tempQuestion: Question[] = [];
+
+        if (category !== {} && category !== null) {
+            question.forEach(x => {
+                if (x.categoryID === category.id || category.id === undefined) {
+                    tempQuestion.push(x);
+                }
+            });
+            return this.filterQuestion(tempQuestion, null, difficultyLevel, matchString);
+        }
+
+        if (difficultyLevel !== null && difficultyLevel !== undefined) {
+            question.forEach(x => {
+                if (x.difficultyLevel === difficultyLevel) {
+                    tempQuestion.push(x);
+                }
+            });
+            return this.filterQuestion(tempQuestion, category, null, matchString);
+        }
+
+        if (matchString.length > 0) {
+            question.forEach(x => {
+                if (x.questionDetail.toLowerCase().includes(matchString.toLowerCase())) {
+                    tempQuestion.push(x);
+                }
+            });
+            return this.filterQuestion(tempQuestion, category, difficultyLevel, '');
+        }
+        return question;
+    }
+    /**
+     * To count the number of Questions
+     */
+    countQuestion() {
+        this.easy = this.medium = this.hard = 0;
+        let questionList = this.filterQuestion(this.question, this.selectedCategory, null, this.matchString);
+        questionList.forEach(x => {
+            switch (x.difficultyLevel) {
+                case 0:
+                    return this.easy++;
```

Comment:
not able to understand that if you are returning to one condition then why foreach is required for it?